containers/widget-server/Dockerfile

```diff
@@ -3,32 +3,50 @@ FROM nginx:1.27.2-alpine-perl
 # Install envsubst
 RUN apk add --no-cache gettext
 
-# Make sure all directories are accessable by the non-root user 
+# Add the nginx configuration
+COPY files/default.conf /etc/nginx/conf.d/
+
+# Make sure all directories are accessible by the non-root user and wrangle
+# files around to enable IPv4-only by duplicating `nginx.conf` and`default.conf`
+# and replacing the `listen` directive.
+#
+# The `/docker-entrypoint.sh` script is also modified to replace `nginx` with `/bin/sh`
+# to allow the container to be started with a read-only file system, run the entry point scripts
+# and use a runtime determined configuration file.
+#
+# IMPORTANT: Copying additional scripts to /etc/nginx/conf.d/ will not work as we nodify
+#            nginx.{ipv4.}conf to only load default.{ipv4.}conf from the conf.d directory.
+#
 # Inspired by https://hub.docker.com/r/nginxinc/nginx-unprivileged
 RUN set -x\
     && chown -R 101:0 /var/cache/nginx \
     && chmod -R g+w /var/cache/nginx \
     && chown -R 101:0 /etc/nginx \
     && chmod -R g+w /etc/nginx \
     && sed -i '/user  nginx;/d' /etc/nginx/nginx.conf \
-    && sed -i 's,/var/run/nginx.pid,/tmp/nginx.pid,' /etc/nginx/nginx.conf
+    && sed -i 's,/var/run/nginx.pid,/tmp/nginx.pid,' /etc/nginx/nginx.conf \
+    && cp /etc/nginx/nginx.conf /etc/nginx/nginx.ipv4.conf \
+    && cp /etc/nginx/conf.d/default.conf /etc/nginx/conf.d/default.ipv4.conf \
+    && sed -i 's/listen \[::\]:8080 ipv6only=off;/listen 8080;/' /etc/nginx/conf.d/default.ipv4.conf \
+    && sed -i 's,include /etc/nginx/conf.d/\*.conf;,include /etc/nginx/conf.d/default.conf;,' /etc/nginx/nginx.conf \
+    && sed -i 's,include /etc/nginx/conf.d/\*.conf;,include /etc/nginx/conf.d/default.ipv4.conf;,' /etc/nginx/nginx.ipv4.conf \
```

Comment:
don't we want to load `/etc/nginx/conf.d/headers/content-security-policy.conf` later in line 39?

Reply:
not an issue because it's `default.conf` that is including that (see [here](https://github.com/nordeck/matrix-widget-toolkit/blob/9d4dadc37dd507f182b829062fdb2cdd8c6a80f1/containers/widget-server/files/default.conf#L28))